pandas/tests/frame/test_analytics.py

```diff
@@ -875,6 +875,12 @@ def test_rank_na_option(self):
         assert_almost_equal(ranks0.values, exp0)
         assert_almost_equal(ranks1.values, exp1)
 
+    def test_rank_axis(self):
+        # check if using axes' names gives the same result
+        df = pd.DataFrame([[2, 1], [4, 3]])
+        assert_frame_equal(df.rank(axis=0), df.rank(axis='index'))
+        assert_frame_equal(df.rank(axis=1), df.rank(axis='columns'))
+
     def test_sem(self):
```

Comment:
can you add onto the rank tests in ``tests/series/test_analytics.py`` simulating the original signature (and asserting an error), e.g.
someone tries. ``s.rank('average')`` e.g. w/o the kw. just for a back-compat that it raises the correct kind of error